### PR TITLE
Added .net core versions of CertificateGeneration project and tests

### PR DESCRIPTION
**Pull request by aaron-schnieder**
Added new dotnet core versions of CertificateGeneration project and tests.
Refactored libraries and methods to remove .net 4.5 library dependencies.
Implemented certificate generation using cross-plat compliant APIs.
Verified all unit tests and integration POST tests run successfully.

### CODE REVIEW

File: services/CertificateGeneration/CertificateGeneration/CertificateGeneration.csproj [hunk 1]
```diff
@@ -156,6 +156,11 @@
         </WebProjectProperties>
       </FlavorProperties>
     </VisualStudio>
+    <MonoDevelop>
+      <Properties>
+        <XspParameters Port="8080" Address="127.0.0.1" SslMode="None" SslProtocol="Default" KeyType="None" CertFile="" KeyFile="" PasswordOptions="None" Password="" Verbose="True" />
```
Comment (logachev) [inline services/CertificateGeneration/CertificateGeneration/CertificateGeneration.csproj]: Is it ok to use 8080?

File: services/CertificateGeneration/CertificateGenerationTests.NetCore/CertificateGenerationTests.NetCore.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  <ItemGroup>
+    <Folder Include="Controllers\" />
+    <Folder Include="Wrappers\" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\CertificateGeneration.NetCore\CertificateGeneration.NetCore.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Remove="UnitTest1.cs" />
```
Comment (logachev) [inline services/CertificateGeneration/CertificateGenerationTests.NetCore/CertificateGenerationTests.NetCore.csproj]: I don't see this file..

File: services/CertificateGeneration/CertificateGenerationTests/CertificateGenerationTests.csproj [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (logachev) [inline services/CertificateGeneration/CertificateGenerationTests/CertificateGenerationTests.csproj]: Note for future: make sure it works for both Mac and Windows.

File: services/CertificateGeneration/CertificateGenerationTests.NetCore/CertificateGenerationTests.NetCore.csproj (hunk 2 above)
Comment (aaron-schnieder) [reply services/CertificateGeneration/CertificateGenerationTests.NetCore/CertificateGenerationTests.NetCore.csproj]: Good catch. Removed.

File: services/CertificateGeneration/CertificateGeneration/CertificateGeneration.csproj (hunk 1 above)
Comment (aaron-schnieder) [reply services/CertificateGeneration/CertificateGeneration/CertificateGeneration.csproj]: Looks like this was auto-added by vs community. Removed.

File: services/CertificateGeneration/CertificateGeneration.NetCore/Wrappers/CertificatesWrapper.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            /* RsaPrivateKeyStructure parameter mapping description found here: https://github.com/neoeinstein/bouncycastle/blob/00ccedf8f4537a8621944f3072453e4603d09c4b/crypto/src/asn1/pkcs/RSAPrivateKeyStructure.cs#L101 */
+
+            try
+            {
+                var privateKey = BouncyCastleUtilities.ToRSA(keyparams);
+                x509 = x509.CopyWithPrivateKey(privateKey);
+            }
+            catch(Exception e) 
+            {
+                Console.WriteLine(e);
```
[the remark this answers is not in the document]
Comment (aaron-schnieder) [reply services/CertificateGeneration/CertificateGeneration.NetCore/Wrappers/CertificatesWrapper.cs]: Yes, good catch. Removed.

File: services/CertificateGeneration/CertificateGeneration.NetCore/Utilities/BouncyCastleUtilities.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+
+namespace CertificateGeneration.Utilities
+{
+    /**
+     * 
+     * Utility methods copied from Org.BouncyCastle.Security.DotNetUtilities class that is not included in BouncyCastle.NetCore package
```
[the remark this answers is not in the document]
Comment (aaron-schnieder) [reply services/CertificateGeneration/CertificateGeneration.NetCore/Utilities/BouncyCastleUtilities.cs]: Good check and yes... https://www.bouncycastle.org/license.html

"Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense..."

**Pull request merged**